File: lib/module/order.dart



import 'dart:convert';

import 'package:appmobile/module/product.dart';

class Order {
  final String id;
  final List<Product> products;
  final List<int> quantity;
  final String address;
  final String userId;
  final String username;
  final int telephone;
  final int orderedAt;
  final int status;
  final double totalPrice;
  final int card;
  final int cv;
final String createdAt;

  Order({
    required this.id,
    required this.products,
    required this.quantity,
    required this.address,
    required this.userId,
    required this.username,
    required this.telephone,
    required this.orderedAt,
    required this.status,
    required this.totalPrice,
    required this.card,
    required this.cv,
    required this.createdAt,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'products': products.map((x) => x.toMap()).toList(),
      'quantity': quantity,
      'address': address,
      'userId': userId,
      'username': username,
      'telephone': telephone,
      'orderedAt': orderedAt,
      'status': status,
      'totalPrice': totalPrice,
      'card': card,
      'cv': cv,
      'createdAt': createdAt,
    };
  }

  factory Order.fromMap(Map<String, dynamic> map) {
    return Order(
      id: map['id'] ?? '',
      products: List<Product>.from(map['products']?.map((x) => Product.fromMap(x))),
      quantity: List<int>.from(map['quantity']),
      address: map['address'] ?? '',
      userId: map['userId'] ?? '',
      username: map['username'] ?? '',
      telephone: map['telephone'] ?? 0,
      orderedAt: map['orderedAt'] ?? 0,
      status: map['status'] ?? 0,
      totalPrice: map['totalPrice']?.toDouble() ?? 0.0,
      card: map['card'],
      cv: map['cv'],
      createdAt:map['createdAt'],
    );
  }

  String toJson() => json.encode(toMap());

  factory Order.fromJson(String source) => Order.fromMap(json.decode(source));
}

File: server/model/order.js
const mongoose = require("mongoose");
const { productSchema } = require("./product");
const paytSchema = require("./payement");

const orderSchema = mongoose.Schema({
  products: [
    {
      product: productSchema,
      quantity: {
        type: Number,
        required: true,
      },
    },
  ],
  totalPrice: {
    type: Number,
    required: true,
  },
  address: {
    type: String,
    required: true,
  },
  telephone: {
    type: Number,
    required: true,
  },
  username: {
    type: String,
    required: true,
  },
  userId: {
    required: true,
    type: String,
  },
  orderedAt: {
    type: Number,
    required: true,
  },
  status: {
    type: Number,
    default: 0,
  },
  card: {
    type: Number,
    required: true,
  },
  cv: {
    type: Number,
    required: true,
  },
  
  createdAt: {
    type: String,

    required: true,

  },});

module.exports = mongoose.model("Order", orderSchema);
